### PR TITLE
Fix for #193

### PR DESCRIPTION
**Pull request by sanjeev-addagarla**
Fix for #193.
https://github.com/greenmail-mail-test/greenmail/issues/193

Enhancement to Support OR term.    
Eg: Users can search Mail box by Subject matching STRING-1 OR STRING-2:         
SearchTerm st = new AndTerm(new SubjectTerm("STRING-1"),new SubjectTerm("STRING-2"));         Message[] messages = emailFolder.search(st);

### CODE REVIEW

File: greenmail-core/src/main/java/com/icegreen/greenmail/imap/commands/SearchCommandParser.java [hunk 1]
```diff
@@ -34,6 +35,8 @@ public SearchTerm searchTerm(ImapRequestLineReader request)
             throws ProtocolException {
         SearchTerm resultTerm = null;
         SearchTermBuilder b = null;
+        SearchKey key = null;
+	boolean Orkey = false;
```
Comment (marcelmay) [inline greenmail-core/src/main/java/com/icegreen/greenmail/imap/commands/SearchCommandParser.java]: Java coding convention : attributes start with lower case and use camel case: "orKey"

File: greenmail-core/src/main/java/com/icegreen/greenmail/imap/commands/SearchCommandParser.java [hunk 2]
```diff
@@ -107,8 +110,27 @@ public SearchTerm searchTerm(ImapRequestLineReader request)
                         negated = false;
                     }
                     b = null;
+					if (key.toString().contains("OR")) {
```
Comment (marcelmay) [inline greenmail-core/src/main/java/com/icegreen/greenmail/imap/commands/SearchCommandParser.java]: Use enum style comparison (more efficient, NPE safe):
SearchKey.OR == key

File: greenmail-core/src/main/java/com/icegreen/greenmail/imap/commands/SearchCommandParser.java [hunk 3]
```diff
@@ -107,8 +110,27 @@ public SearchTerm searchTerm(ImapRequestLineReader request)
...
+						if (SearchKey.OR == key) {
```
Comment (marcelmay) [inline greenmail-core/src/main/java/com/icegreen/greenmail/imap/commands/SearchCommandParser.java]: Did previous line 113 not just check if key is SearchKey.OR?

File: greenmail-core/src/main/java/com/icegreen/greenmail/imap/commands/SearchCommandParser.java [hunk 4]
```diff
@@ -107,8 +110,27 @@ public SearchTerm searchTerm(ImapRequestLineReader request)
...
+						resultTerm = resultTerm == null ? searchTerm : new OrTerm(resultTerm, searchTerm);
+						Orkey = true;
+						}
+
+					} else {
+						if (Orkey) {
+							if (key.toString().contains("ALL")) {
```
Comment (marcelmay) [inline greenmail-core/src/main/java/com/icegreen/greenmail/imap/commands/SearchCommandParser.java]: Do you want to match SearchKey.ALL? Use enum style.

File: greenmail-core/src/main/java/com/icegreen/greenmail/imap/commands/SearchCommandParser.java [hunk 5]
```diff
@@ -107,8 +110,27 @@ public SearchTerm searchTerm(ImapRequestLineReader request)
...
                     resultTerm = resultTerm == null ? searchTerm : new AndTerm(resultTerm, searchTerm);
-                }
+
+							} else {
+								resultTerm = resultTerm == null ? searchTerm : new OrTerm(resultTerm, searchTerm);
+
```
Comment (marcelmay) [inline greenmail-core/src/main/java/com/icegreen/greenmail/imap/commands/SearchCommandParser.java]: Remove empty line (applies to other changed lines, too).

**Comment by sanjeev-addagarla** [comment]
Updated as suggested

**Comment by marcelmay** [comment]
Hi @sanjeev-addagarla ,

thx alot for the updates. Can you run the tests?

E.g. on greenmail-core
`mvn clean test -Dtest=ImapSearchTest` ?

**Comment by technotronic12** [comment]
I've pushed my changes

**Comment by marcelmay** [comment]
There are still some merge conflicts, so you would have to rebase.

Ok, I'll resolve these.

**Pull request merged**